project/Build.scala

```diff
@@ -1180,9 +1209,21 @@ object Build {
     case Bootstrapped => scaladoc
   }
 
-  lazy val `scaladoc-testcases` = project.in(file("scaladoc-testcases")).asScaladocTestcases
+  lazy val `scaladoc-testcases` = project.in(file("scaladoc-testcases")).asScaladocTestcases(Bootstrapped)
+  lazy val `scaladoc-testcases-nonBootstrapped` = project.in(file("scaladoc-testcases")).asScaladocTestcases(NonBootstrapped)
+
+  def scalaDocTestcases(implicit mode: Mode): Project = mode match {
+    case NonBootstrapped => `scaladoc-testcases-nonBootstrapped`
+    case Bootstrapped => `scaladoc-testcases`
+  }
+
+  lazy val `scaladoc-js` = project.in(file("scaladoc-js")).asScaladocJs(Bootstrapped)
+  lazy val `scaladoc-js-nonBootstrapped` = project.in(file("scaladoc-js")).asScaladocJs(NonBootstrapped)
```

Comment:
Is scaladoc-js-nonBootstrapped actually needed for anything? Otherwise you could just keep the bootstrapped version and avoid some complexity in the build.

Reply:
We need to compile scala-js parts of `scaladoc-nonBootstraped`. Its output will be used in documentation shipped to Maven Central.

Reply:
I'm not sure I understand, we never publish non-bootstrapped projects on maven central.

Reply:
`scaladoc-js` compiles into javascript code that is included in the documentation generated by `scaladoc` (implementation of search etc.). In order to document e.g. `scala3-library-bootstrapped` we need to use not bootstrapped `scaladoc` and hence we need non bootstrapped version of scaladoc-js to add all the javascript pieced to documentation.

If there is better way to implement this I am happy to change it after RC2 is released.

Reply:
>  In order to document e.g. scala3-library-bootstrapped we need to use not bootstrapped scaladoc

Oh I see, this is tricky indeed but it might be possible to avoid a cycle if we can do things in this order: `compile scala3-library-bootstrapped -> compile scaladoc-js -> use this scaladoc-js to generate the scala3-library-bootstrapped docs`, but that could be done post-RC2 indeed.